### PR TITLE
Improve error message when an argument is not provided

### PR DESCRIPTION
**Pull request by joelverhagen**
`KeyNotFoundException` can be thrown because this method is used a lot:
https://github.com/NuGet/ServerCommon/blob/01f97075c87121e48abd428771cb4f758861d572/src/NuGet.Services.Configuration/DictionaryExtensions.cs#L38

It's better to show the error message here and helps local/ops execution of `ng.exe`.

### CODE REVIEW

File: src/Ng/Program.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
```
Comment (agr) [inline src/Ng/Program.cs]: Why is this needed?

File: src/Ng/Program.cs [hunk 2]
```diff
@@ -92,7 +93,7 @@ public static async Task MainAsync(string[] args)
                 job = NgJobFactory.GetJob(jobName, telemetryService, loggerFactory);
                 await job.RunAsync(arguments, cancellationTokenSource.Token);
             }
-            catch (ArgumentException ae)
+            catch (Exception ae) when (ae is ArgumentException || ae is KeyNotFoundException)
             {
                 _logger?.LogError("A required argument was not found or was malformed/invalid: {Exception}", ae);
```
Comment (zhhyu) [inline src/Ng/Program.cs]: Maybe improve the log message to denote that the key is not found, or another catch block for this case.

File: src/Ng/Program.cs (hunk 1 above)
Comment (joelverhagen) [reply src/Ng/Program.cs]: `KeyNotFoundException` is in this namespace:
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.keynotfoundexception?view=netframework-4.8

File: src/Ng/Program.cs (hunk 2 above)
Comment (joelverhagen) [reply src/Ng/Program.cs]: Good idea, I'll do that.

File: src/Ng/Program.cs [hunk 3]
```diff
@@ -92,7 +93,7 @@ public static async Task MainAsync(string[] args)
                 job = NgJobFactory.GetJob(jobName, telemetryService, loggerFactory);
                 await job.RunAsync(arguments, cancellationTokenSource.Token);
             }
-            catch (ArgumentException ae)
+            catch (Exception ae) when (ae is ArgumentException || ae is KeyNotFoundException)
```
Comment (agr) [inline src/Ng/Program.cs]: In general, catching an `ArgumentException` here, and representing it as a "required argument id missing/invalid" is a bit of a stretch, since the `RunAsync` call is tried that runs complete job. `ArgumentException` can be thrown anywhere, not only at the argument validation level.
Ideally, we'd probably want a separate command line argument validation/consumption stage separate from running the job, but I'd imagine that would be quite out of scope for the intended change.

Comment (joelverhagen) [reply src/Ng/Program.cs]: Yup, I didn't want to fix the questionable design of the command line parsing -- I just wanted to make the current mess more usable.
